### PR TITLE
cleanup dependencies

### PR DESCRIPTION
**Pull request by adecaro**
- remove dependency to libp2p from packages that should not have it
- fabric dependencies from the view platform

### CODE REVIEW

File: platform/view/services/comm/pki.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		return nil
+	}
+}
+
+func (p PKISupport) PublicKeyID(key any) []byte {
+	switch d := key.(type) {
+	case *ecdsa.PublicKey:
+		raw, err := x509.MarshalPKIXPublicKey(d)
+		if err != nil {
+			return nil
```
Comment (alexandrosfilios) [inline platform/view/services/comm/pki.go:49]: Should we log the error or is it expected that it happens?

File: platform/view/services/comm/pki.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/view"
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+type PKISupport struct{}
+
+func (p PKISupport) ExtractPublicKey(id view.Identity) any {
+	certRaw, _ := pem.Decode(id)
+	switch {
```
Comment (alexandrosfilios) [inline platform/view/services/comm/pki.go:24]: Maybe replacing the switch statement with a nil check at the beginning could be more readable:
```
certRaw, _ := pem.Decode(id)
if certRaw == nil {
 return nil
}
cert, err := x509.ParseCertificate(certRaw.Bytes)
...
```

**Review by alexandrosfilios — APPROVED** [review]

File: platform/view/services/hash/hashable.go [hunk 3]
```diff
@@ -13,6 +13,21 @@ import (
 
 type Hashable []byte
 
+func (id Hashable) Raw() []byte {
```
[the remark this answers is not in the document]
Comment (adecaro) [reply platform/view/services/hash/hashable.go:16]: 👍

File: platform/view/services/comm/pki.go (hunk 1 above)
Comment (adecaro) [reply platform/view/services/comm/pki.go:49]: good point 👍

File: platform/view/services/comm/pki.go (hunk 2 above)
Comment (adecaro) [reply platform/view/services/comm/pki.go:24]: 👍

**Pull request merged**